### PR TITLE
Eiger

### PR DESCRIPTION
**Pull request by JunAishima**
extract Eiger-related objects from profile_collection, add setup functions (same order of PVs as current LSDC)

### CODE REVIEW

File: mxtools/scans.py [hunk 1]
```diff
@@ -86,3 +94,65 @@ def setup_vector_program(num_images, angle_start, angle_end, exposure_period_per
...
+    yield from bps.mv(eiger.file_owner, getpass.getuser())
+    yield from bps.mv(eiger.file_owner_grp, grp.getgrgid(os.getgid())[0])
+    yield from bps.mv(eiger.file_perms, 420)
+    file_prefix_minus_directory = str(file_prefix)
+    try:
+        file_prefix_minus_directory = file_prefix_minus_directory[
+            file_prefix_minus_directory.rindex("/") + 1 : len(file_prefix_minus_directory)  # noqa: E203
+        ]
+    except ValueError:
+        pass
```
Comment (mrakitin) [inline mxtools/scans.py]:
```suggestion
    file_prefix_minus_directory = file_prefix_minus_directory.split('/')[-1]
```

File: mxtools/scans.py [hunk 2]
```diff
@@ -86,3 +94,65 @@ def setup_vector_program(num_images, angle_start, angle_end, exposure_period_per
...
+
+    yield from bps.mv(eiger.acquire_time, exposure_per_image)
+    yield from bps.mv(eiger.acquire_period, exposure_per_image)
+    yield from bps.mv(eiger.num_images, num_images)
+    yield from bps.mv(eiger.file_path, data_directory_name)
+    yield from bps.mv(eiger.fw_name_pattern, f"{file_prefix_minus_directory}_$id")
+    yield from bps.mv(eiger.sequence_id, file_number_start)
```
Comment (mrakitin) [inline mxtools/scans.py:136]: I am not sure if those 3 parameters will be overwritten by `eiger.stage()`... I guess we need to create our custom method to overwrite https://github.com/bluesky/ophyd/blob/d0bc07ef85cf2d69319a059485a21a5cdbd0e677/ophyd/areadetector/filestore_mixins.py#L403-L422 (uuid4-based file names).

**Review by mrakitin — COMMENTED** [review]
I have no major issues with that, I have a single suggestion to make the splitting more readable. Also, for the future updates, we may consider overwriting [`make_filename`](https://github.com/bluesky/ophyd/blob/d0bc07ef85cf2d69319a059485a21a5cdbd0e677/ophyd/areadetector/filestore_mixins.py#L403-L422).

File: mxtools/scans.py (hunk 2 above)
Comment (JunAishima) [reply mxtools/scans.py:136]: after our conversation, I agree with what you are proposing. we agree to implement this custom method in a later PR

**Pull request merged**